### PR TITLE
feat(trash): add synchronized trash support for windows

### LINKED ISSUE #2331: Trash files on Windows as well as Unix system
**Is this a question?**
Move to Windows recycle bin instead of deleting files completely as I need to delete files. I know there is a function 'Trash' with shortcut `D`, but it told me:

> Trash is currently a UNIX only feature!

**Can this functionality be implemented utilising API?**
Sorry I'm not good at it and I don't know about it... But I know there's a PowerShell script can do that:

```powershell
$sh = new-object -comobject "Shell.Application"
$ns = $sh.Namespace(0).ParseName("PATH\TO\FILE\TO\DELETE")
$ns.InvokeVerb("delete")
```

**Is your feature request related to a problem? Please describe.**
Add Windows support of command 'Trash'.

**Describe the solution you'd like**
Add Windows support of command 'Trash'.

**Describe alternatives you've considered**
#279 Shows the way I used it on WSL.

**Additional context**
Nothing.

### PR DESCRIPTION
**Pull request by linrongbin16**
fixes #2331

Support trash-cli on Windows, relate to discussion https://github.com/nvim-tree/nvim-tree.lua/discussions/2333 and #2331.

1. remove `Unix only` from trash action.
2. set `detach=false` and `jobwait { job }` on Windows to make trash operation sync on Windows. This is because: in my testing/practicing, when set `detach=true` in jobstart, nvim (running in Windows Terminal app) quickly switch to desktop, then switch back to nvim, that makes the screen blink. I guess the **process detech** behaviour is different under Windows, to avoid this blink I set sync wait under Windows.
3. manual tested on Windows 10 x86_64, nvim v0.9.1, latest nvim-tree master.

### CODE REVIEW

**Comment by linrongbin16** [comment]
I mark this PR as draft, and try to support async.

let me take time to test the jobstart api.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Looking good.

I'm happy with the sync: cautious but reliable behaviour is preferred. We don't need any more options.

What is the actual `trash.cmd`? We can set this for windows, as I have just done for macos: https://github.com/nvim-tree/nvim-tree.lua/pull/2336
It might be easiest to merge that PR before this one.

Please:
- [x] update help section 10
- [x] update help section 4 nvim-tree.trash.cmd
- [x] test under WSL and powershell (if you haven't already)

**Comment by linrongbin16** [comment]
hi @alex-courtis ,

 1. the actual trash.cmd is `trash` (see https://github.com/sindresorhus/trash-cli), the two implementations (python and nodejs) use the same cmd name `trash`. so no other things need to change.
 2. as far as I know, these two trash cli implementations are most popular, and they all use `trash`, so it should be good to set default trash cmd to `trash`.
 3. todo: add help doc and other testing

File: lua/nvim-tree/actions/fs/trash.lua [hunk 1]
```diff
@@ -30,16 +30,11 @@ function M.fn(node)
   end
 
   -- configs
-  if utils.is_unix then
```
Comment (alex-courtis) [inline lua/nvim-tree/actions/fs/trash.lua:33]: Can we please be conservative and use is_unix or is_windows ?

We'll sort out macos later: #2336

**Comment by alex-courtis** [comment]
>     4. [ ]  test under WSL. -- Is this [WSL](https://learn.microsoft.com/en-us/windows/wsl/install#upgrade-version-from-wsl-1-to-wsl-2) or WSL2? I don't have WSL on my Windows 10 or Windows 11, because it's quite legacy, I only have WSL2, which is actually a ubuntu VM running on Windows 10 host.

RE WSL or WSL2 I'll leave that to you... I'm not familiar with windows.

A quick test on whatever is the current WSL would  be great.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Looking good, many thanks for your contribution.

**Comment by linrongbin16** [comment]
> A quick test on whatever is the current WSL would be great.

Test on WSL2 is done.

File: lua/nvim-tree/actions/fs/trash.lua (hunk 1 above)
Comment (linrongbin16) [reply lua/nvim-tree/actions/fs/trash.lua:33]: update: use `is_unix or is_windows` check

Comment (linrongbin16) [reply lua/nvim-tree/actions/fs/trash.lua:33]: please check again, should be OK now.

**Review by alex-courtis — APPROVED** [review]
Many thanks for your contribution!

**Pull request merged**